kms/keysource.go

```diff
@@ -162,37 +161,56 @@ func (key MasterKey) createSession() (*session.Session, error) {
 }
 
 // ToMap converts the MasterKey to a map for serialization purposes
-func (key MasterKey) ToMap() map[string]string {
-	out := make(map[string]string)
+func (key MasterKey) ToMap() map[string]interface{} {
+	out := make(map[string]interface{})
 	out["arn"] = key.Arn
 	if key.Role != "" {
 		out["role"] = key.Role
 	}
 	out["created_at"] = key.CreationDate.UTC().Format(time.RFC3339)
 	out["enc"] = key.EncryptedKey
 	if key.EncryptionContext != nil {
-		var outContexts []string
+		outcontext := make(map[string]string)
 		for k, v := range key.EncryptionContext {
-			outContexts = append(outContexts, k+":"+*v)
+			outcontext[k] = *v
 		}
-		sort.Strings(outContexts)
-		out["context"] = strings.Join(outContexts, ",")
+		out["context"] = outcontext
 	}
 	return out
 }
 
-// ParseKMSContext takes a comma-separated list of KMS context key:value pairs and returns a map
-func ParseKMSContext(in string) map[string]*string {
-	if in == "" {
-		return nil
-	}
+// ParseKMSContext takes either a KMS context map or a comma-separated list of KMS context key:value pairs and returns a map
+func ParseKMSContext(in interface{}) map[string]*string {
 	out := make(map[string]*string)
-	for _, kv := range strings.Split(in, ",") {
-		kv := strings.Split(kv, ":")
-		if len(kv) != 2 {
+
+	switch in := in.(type) {
+	case map[string]interface{}:
+		if len(in) == 0 {
+			return nil
+		}
+		for k, v := range in {
+			var v = v.(string)
```

Comment:
When casting, I would prefer it if you checked for an error and showed a warning, otherwise, this will panic. It should never happen, because this should never be manipulated by the user and a non-string should never wind up there. There are a bunch of unchecked casts in the codebase already, so I understand you might be copying our style, but in this case, you really shouldn't 😉 

Something like:

```go
value, ok := v.(string)
if !ok {
    fmt.Printf("[WARNING]: KMS Encryption Context contains a non-string key, context will not be used")
    return nil
}
```